AudioContext::resume not `Send` anymore?

I have this error with `AudioContext::resume` after upgrading to 1.0.0-rc.6

```
error[E0277]: `std::sync::MutexGuard<'_, Box<dyn web_audio_api::io::AudioBackendManager>>` cannot be sent between threads safely
    --> src/audio_context.rs:177:13
     |
177  |       ctx.env.execute_tokio_future(
     |               ^^^^^^^^^^^^^^^^^^^^ `std::sync::MutexGuard<'_, Box<dyn web_audio_api::io::AudioBackendManager>>` cannot be sent between threads safely
178  | /         async move {
179  | |             context_clone.resume().await;
180  | |             Ok(())
181  | |         },
     | |_________- within this `{async block@src/audio_context.rs:178:9: 181:10}`
     |
     = help: within `{async block@src/audio_context.rs:178:9: 181:10}`, the trait `Send` is not implemented for `std::sync::MutexGuard<'_, Box<dyn web_audio_api::io::AudioBackendManager>>`, which is required by `{async block@src/audio_context.rs:178:9: 181:10}: Send`
     = note: required because it captures the following types: `std::sync::MutexGuard<'_, Box<dyn web_audio_api::io::AudioBackendManager>>`, `futures_channel::oneshot::Receiver<()>`
note: required because it's used within this `async` fn body
    --> src/index.crates.io-6f17d22bba15001f/web-audio-api-1.0.0-rc.6/src/context/online.rs:474:32
     |
474  |       pub async fn resume(&self) {
     |  ________________________________^
475  | |         // Lock the backend manager mutex to avoid concurrent calls
476  | |         log::debug!("Resume called, locking backend manager");
477  | |         let backend_manager_guard = self.backend_manager.lock().unwrap();
...    |
500  | |         log::debug!("Resumed audio graph");
501  | |     }
     | |_____^
     = note: required because it captures the following types: `impl Future<Output = ()>`
note: required because it's used within this `async` block
    --> src/audio_context.rs:178:9
     |
178  | /         async move {
179  | |             context_clone.resume().await;
180  | |             Ok(())
181  | |         },
     | |_________^
note: required by a bound in `napi::Env::execute_tokio_future`
    --> src/index.crates.io-6f17d22bba15001f/napi-2.16.4/src/env.rs:1075:18
     |
1072 |   pub fn execute_tokio_future<
     |          -------------------- required by a bound in this associated function
...
1075 |     F: 'static + Send + Future<Output = Result<T>>,
     |                  ^^^^ required by this bound in `Env::execute_tokio_future`
```

I imagine this is due to the fixes done in #499 but I can't figure out the problem

My wrapping code (which didn't change since previous version) is:

```rs
#[js_function]
fn resume(ctx: CallContext) -> Result<JsObject> {
    let js_this = ctx.this_unchecked::<JsObject>();
    let napi_context = ctx.env.unwrap::<NapiAudioContext>(&js_this)?;
    let context_clone = Arc::clone(&napi_context.0);

    ctx.env.execute_tokio_future(
        async move {
            context_clone.resume().await;
            Ok(())
        },
        |&mut env, _val| env.get_undefined(),
    )
}
```

--- a/src/context/offline.rs
+++ b/src/context/offline.rs
@@ -630,4 +630,15 @@ mod tests {
 
         assert!(complete.load(Ordering::Relaxed));
     }
+
+    fn require_send_sync<T: Send + Sync>(_: T) {}
+
+    #[test]
+    fn test_all_futures_thread_safe() {
+        let context = OfflineAudioContext::new(2, 555, 44_100.);
+
+        require_send_sync(context.start_rendering());
+        require_send_sync(context.suspend(1.));
+        require_send_sync(context.resume());
+    }
 }

--- a/src/context/online.rs
+++ b/src/context/online.rs
@@ -470,29 +470,30 @@ impl AudioContext {
     ///
     /// * The audio device is not available
     /// * For a `BackendSpecificError`
-    #[allow(clippy::await_holding_lock)] // false positive due to explicit drop
     pub async fn resume(&self) {
-        // Lock the backend manager mutex to avoid concurrent calls
-        log::debug!("Resume called, locking backend manager");
-        let backend_manager_guard = self.backend_manager.lock().unwrap();
-
-        if self.state() != AudioContextState::Suspended {
-            log::debug!("Resume no-op - context is not suspended");
-            return;
-        }
-
-        // Ask the audio host to resume the stream
-        backend_manager_guard.resume();
-
-        // Then, ask to resume rendering via a control message
-        log::debug!("Resumed audio stream, waking audio graph");
         let (sender, receiver) = oneshot::channel();
-        let notify = OneshotNotify::Async(sender);
-        self.base
-            .send_control_msg(ControlMessage::Resume { notify });
 
-        // Drop the Mutex guard so we won't hold it across an await
-        drop(backend_manager_guard);
+        {
+            // Lock the backend manager mutex to avoid concurrent calls
+            log::debug!("Resume called, locking backend manager");
+            let backend_manager_guard = self.backend_manager.lock().unwrap();
+
+            if self.state() != AudioContextState::Suspended {
+                log::debug!("Resume no-op - context is not suspended");
+                return;
+            }
+
+            // Ask the audio host to resume the stream
+            backend_manager_guard.resume();
+
+            // Then, ask to resume rendering via a control message
+            log::debug!("Resumed audio stream, waking audio graph");
+            let notify = OneshotNotify::Async(sender);
+            self.base
+                .send_control_msg(ControlMessage::Resume { notify });
+
+            // Drop the Mutex guard so we won't hold it across an await point
+        }
 
         // Wait for the render thread to have processed the resume message
         // The AudioContextState will be updated by the render thread.
@@ -766,5 +767,20 @@ mod tests {
 
         let time5 = context.current_time();
         assert_eq!(time5, time4); // no progression of time
+    }
+
+    fn require_send_sync<T: Send + Sync>(_: T) {}
+
+    #[test]
+    fn test_all_futures_thread_safe() {
+        let options = AudioContextOptions {
+            sink_id: "none".into(),
+            ..AudioContextOptions::default()
+        };
+        let context = AudioContext::new(options);
+
+        require_send_sync(context.suspend());
+        require_send_sync(context.resume());
+        require_send_sync(context.close());
     }
 }

